### PR TITLE
wxDCTextBgColourChanger and wxDCTextBgModeChanger helper classes

### PR DESCRIPTION
**Pull request by vgromov**
wxDCTextBgColourChanger and wxDCTextBgModeChanger helper classes added and documented

### CODE REVIEW

File: include/wx/dc.h [hunk 1]
```diff
@@ -1405,6 +1405,78 @@ class WXDLLIMPEXP_CORE wxDCTextColourChanger
...
+// ----------------------------------------------------------------------------
+// helper class: you can use it to temporarily change the DC text background colour and
+// restore it automatically when the object goes out of scope
+// ----------------------------------------------------------------------------
+
+class WXDLLIMPEXP_CORE wxDCTextBgColourChanger
+{
+public:
+    wxDCTextBgColourChanger(wxDC& dc) : m_dc(dc), m_colBgOld() { }
+
```
Comment (PBfordev) [inline include/wx/dc.h]: Calling `m_colBgOld()` makes no difference and does no have to be here: the default ctor would be called automatically anyway.

File: include/wx/dc.h [hunk 2]
```diff
@@ -1405,6 +1405,78 @@ class WXDLLIMPEXP_CORE wxDCTextColourChanger
...
+    wxDECLARE_NO_COPY_CLASS(wxDCTextBgColourChanger);
+};
+
+// ----------------------------------------------------------------------------
+// helper class: you can use it to temporarily change the DC text background mode and
+// restore it automatically when the object goes out of scope
+// ----------------------------------------------------------------------------
+
+class WXDLLIMPEXP_CORE wxDCTextBgModeChanger
```
Comment (PBfordev) [inline include/wx/dc.h]: Is the class name intuitive enough, the wrapped `wxDC` method does not have _Text_ in its name nor are any of the words abbreviated in the changer names or `wxDC` method names (this applies also to [wxDCTextBgColourChanger](url))? 

OTOH, according to the documentation, the mode is used only for drawing text and perhaps the full names would be too long for some...

File: include/wx/dc.h [hunk 3]
```diff
@@ -1405,6 +1405,78 @@ class WXDLLIMPEXP_CORE wxDCTextColourChanger
...
+    wxDCTextBgModeChanger(wxDC& dc) : m_dc(dc), m_modeOld(wxBRUSHSTYLE_INVALID) { }
+
+    wxDCTextBgModeChanger(wxDC& dc, int mode) : m_dc(dc)
+    {
+        Set(mode);
+    }
+
+    ~wxDCTextBgModeChanger()
+    {
+        if ( wxBRUSHSTYLE_INVALID != m_modeOld )
```
Comment (PBfordev) [inline include/wx/dc.h]: I believe wxWidgets does not use coding style with the constant on the left in conditions of `if` statements. 

Applies also to `Set()`.

File: interface/wx/dc.h [hunk 4]
```diff
@@ -1842,6 +1842,113 @@ class wxDCTextColourChanger
...
 
 
+/**
+    @class wxDCTextBgColourChanger
+
+    wxDCTextBgColourChanger is a small helper class for setting a background
+    text colour on a wxDC and unsetting it automatically in the destructor,
+    restoring the previous one.
+
+    @library{wxcore}
```
Comment (PBfordev) [inline interface/wx/dc.h]: The documentation for both classes is missing `@since 3.1.1` tag.

File: interface/wx/dc.h [hunk 5]
```diff
@@ -1842,6 +1842,113 @@ class wxDCTextColourChanger
...
+    @category{gdi}
+
+    @see wxDC::SetTextBackground(), wxDCFontChanger, wxDCPenChanger, wxDCBrushChanger,
```
Comment (PBfordev) [inline interface/wx/dc.h]: Perhaps references to`wxDCTextBgModeChanger`  and `wxDCTextColourChanger` could be added here as well (and perhaps references to classes not related to drawing text - `wxDCPenChanger`, `wxDCBrushChanger`, `wxDCClipper` - removed if you feel there is too many references)?

File: interface/wx/dc.h [hunk 6]
```diff
@@ -1842,6 +1842,113 @@ class wxDCTextColourChanger
...
+        anything in its destructor unless Set() is called -- in which case it
+        would reset the previous mode.
+     */
+    wxDCTextBgModeChanger(wxDC& dc);
+
+    /**
+        Sets @a mode on the given @a dc, storing the old one.
+
+        @param dc
+            The DC where the colour must be temporary set.
```
Comment (PBfordev) [inline interface/wx/dc.h]: Please change _colour_ to _mode_.

File: interface/wx/dc.h [hunk 7]
```diff
@@ -1842,6 +1842,113 @@ class wxDCTextColourChanger
...
+    @class wxDCTextBgModeChanger
+
+    wxDCTextBgModeChanger is a small helper class for setting a background
+    text mode on a wxDC and unsetting it automatically in the destructor,
+    restoring the previous one.
+
+    @library{wxcore}
+    @category{gdi}
+
+    @see wxDC::SetBackgroundMode(), wxDCFontChanger, wxDCPenChanger, wxDCBrushChanger,
```
Comment (PBfordev) [inline interface/wx/dc.h]: Please consider adding references to other text drawing related changers, see the previous comment.

**Review by PBfordev — COMMENTED** [review]
Sorry for being so detailed in the review.

Now perhaps the only commonly would-be-used changer missing is the one for the background brush. I once considered adding it myself but perhaps that could be one changers too many and another solution is preferable...

**Comment by vadz** [comment]
IIUC this was replaced by #1298 (@vgromov please either update the existing PRs or close them if you open new ones replacing them, otherwise things get rather confusing).

**Pull request closed without merging**